src/App.tsx

```diff
@@ -14,11 +14,75 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGoods: string[]
+}
+class App extends React.Component {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  selectGood = (newGood:string) => {
+    const { selectedGoods } = this.state;
+
+    if (selectedGoods.includes(newGood)) {
+      selectedGoods.splice(selectedGoods.indexOf(newGood), 1);
+    } else {
+      selectedGoods.push(newGood);
+    }
+
+    this.setState({});
+  };
+
+  clearList = () => {
+    this.state.selectedGoods = [];
+    this.setState({});
```

Comment:
Так не можна оновлювати стейт
```suggestion
    this.setState({ selectedGoods: [] });
```